public/src/managers/CourseManager.ts

```diff
@@ -338,7 +338,7 @@ export class CourseManager {
         if (!studentCourse.enrollment) {
             return;
         }
-        if (!assignments) {
+        if (!assignments || assignments?.length < 1) {
```

Comment:
I don't think you need to protect the second clause with a nil check, since the first clause already does that and will short-circuit the second clause. But it's fine to be safe. Anyway, maybe you should apply a similar fix to `fillLinksGroup`.

Reply:
Right. Fixed it.